pkg/sentry/fsimpl/fuse/connection.go

```diff
@@ -77,6 +77,12 @@ type futureResponse struct {
 type Connection struct {
 	fd *DeviceFD
 
+	// Lock protect access to struct memeber
+	Lock sync.Mutex
```

Comment:
It would be better if can specify which subset of variables to protect? A global lock seems unnecessary

Reply:
In Linux, this is the only lock on connection, and it is specified to protect everything, so I guess we should use this for all struct members. However, I do notice there is another `initializedLock` in connection struct, I'm not sure where that comes from, but I would suggest replace that one with this.

Reply:
Yes and no. It will not protect the variables of which data racing is safe. `initializedLock` is to make sure all CPU cores (with local cache) see the initialize flag before proceeding. Ideally the queuing logic (queue the request before the FUSE_INIT reply is received) should not worry about that sync